archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java

```diff
@@ -151,4 +154,28 @@ private GeneralCodingRules() {
     @PublicAPI(usage = ACCESS)
     public static final ArchRule NO_CLASSES_SHOULD_USE_JODATIME =
             noClasses().should(USE_JODATIME).because("modern Java projects use the [java.time] API instead");
+
+    /**
+     * For information about checking this condition, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchCondition<JavaField> USE_FIELD_INJECTION = useFieldInjection().as("use field injection");
+
+    private static ArchCondition<JavaField> useFieldInjection() {
+        ArchCondition<JavaField> annotatedWithAutowired = beAnnotatedWith("org.springframework.beans.factory.annotation.Autowired");
+        ArchCondition<JavaField> annotatedWithValue = beAnnotatedWith("org.springframework.beans.factory.annotation.Value");
+        ArchCondition<JavaField> annotatedWithInject = beAnnotatedWith("javax.inject.Inject");
+        ArchCondition<JavaField> annotatedWithResource = beAnnotatedWith("javax.annotation.Resource");
+        return annotatedWithAutowired.or(annotatedWithValue).or(annotatedWithInject).or(annotatedWithResource);
+    }
+
+    /**
+     * Field injection is seen as an anti-pattern.
+     * It is a good practice to use constructor injection for mandatory dependencies and setter injection for optional dependencies.
+     * <br>
+     * For information about checking this rule, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchRule NO_CLASSES_SHOULD_USE_FIELD_INJECTION =
```

Comment:
The text of the rule and field doesn't really match now, right? I would either call both "no fields should be injected" or "no classes should use field injection", but now the field name is "no classes..." but the rule text will then be "no fields..." or am I overlooking something here? Also "no fields should use field injection" sounds a little strange to me :wink: